rustls-platform-verifier/src/tests/verification_mock/mod.rs

```diff
@@ -258,6 +264,12 @@ mock_root_test_cases! {
         chain: &[ROOT1_INT1_EXAMPLE_COM_GOOD, ROOT1_INT1],
         stapled_ocsp: None,
         verification_time: verification_time(),
+        #[cfg(not(any(target_vendor = "apple", windows)))]
+        expected_result: Err(TlsError::InvalidCertificate(CertificateError::NotValidForNameContext {
+            expected: ServerName::DnsName(DnsName::try_from("example.org").unwrap()),
+            presented: vec!["DnsName(\"example.com\")".to_owned()]
+        })),
+        #[cfg(any(target_vendor = "apple", windows))]
         expected_result: Err(TlsError::InvalidCertificate(CertificateError::NotValidForName)),
```

Comment:
Would you mind opening a followup issue to investigate if we can improve the error reporting of invalid names for the non-webpki based verifiers?

Reply:
Will do.

Reply:
Oh TIL GitHub added a "create followup issue button". I went ahead and made it myself to save you the hassle.

Reply:
Yeah, sorry for the ordering weirdness. GitHub didn't refresh in your "Will do." comment for me until after this merged and I refreshed the page.